ntp-proto/src/filter.rs

```diff
@@ -366,25 +366,32 @@ struct SurvivorTuple<'a> {
 /// Collect the candidates within the correctness interval
 #[allow(dead_code)]
 fn construct_survivors<'a>(
-    chime_list: &[CandidateTuple<'a>],
+    chime_list: &'a [CandidateTuple<'a>],
```

Comment:
So now this is allocating a `Vec<SurvivorTuple<'a>>`. I think you could just skip the `collect()` and return `imp lIterator<SurvivorTuple<'a>> + 'a` instead? (Unless there are other uses for `find_interval() and `filter_survivor()`, I would probably merge all three functions into one.)

Reply:
we need these in a vector because we'll be removing (iteratively filtering) elements from that vector (see the next PR)